src/Microsoft.PowerShell.ConsoleHost/host/msh/CommandLineParameterParser.cs

```diff
@@ -509,6 +516,19 @@ private void ParseHelper(string[] args)
                     switchKey = switchKey.Substring(1);
                 }
 
+                // If version is in the commandline, don't continue to look at any other parameters
```

Comment:
@vors, the @PowerShell/powershell-committee discussed this and that is pretty much the thinking: `-version` for Windows PowerShell really only supported 2.0 and we're moving away from that.  Aligning with Linux convention makes sense.  PSCore supports side-by-side intrinsically and using file paths to launch different versions is completely acceptable.